src/main/java/org/datadog/jenkins/plugins/datadog/clients/ClientFactory.java

```diff
@@ -62,6 +62,9 @@ public static DatadogClient getClient() {
             targetPort = descriptor.getTargetPort();
             targetLogCollectionPort = descriptor.getTargetLogCollectionPort();
         }
+        if (targetApiKey == null) {
```

Comment:
Without a valid API key it does not make sense to return a client.
However, I now see this affects many callers, I will reduce the area to async period worker